File: python/figure.py
import math
from abc import ABC, abstractmethod

class Figure(ABC):
    def __init__(self, color: str):
        self.color = color

    def getColor(self):
        return self.color

    @abstractmethod
    def perimeter(self):
        pass

    @abstractmethod
    def area(self):
        pass

File: java/figure/src/figure/Figure.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package figure;

/**
 *
 * @author eyro1
 */
abstract class Figure {
    protected String color;

    public Figure(String color) {
        this.color = color;
    }

    public String getColor() {
        return color;
    }

    public abstract double perimeter();

    public abstract double area();
    
    public abstract String getName();
}
